src/components/expanded-state/chart/chart-data-labels/ChartPriceLabel.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback } from 'react';
+import { Text } from '@/design-system';
+import { useAccountSettings } from '@/hooks';
+import { ChartYLabel } from '@/react-native-animated-charts/src';
+import { SupportedCurrency, supportedNativeCurrencies } from '@/references';
+import { DEVICE_WIDTH } from '@/utils/deviceUtils';
+import { orderOfMagnitudeWorklet, significantDecimalsWorklet } from '@/safe-math/SafeMath';
+
+function calculateDecimalPlaces({
+  amount,
+  minimumDecimals = 0,
+  maximumDecimals = 6,
+  precisionAdjustment,
+}: {
+  amount: number | string;
+  minimumDecimals?: number;
+  maximumDecimals?: number;
+  precisionAdjustment?: number;
+}): {
+  minimumDecimalPlaces: number;
+  maximumDecimalPlaces: number;
+} {
+  'worklet';
+  const orderOfMagnitude = orderOfMagnitudeWorklet(amount);
+  const significantDecimals = significantDecimalsWorklet(amount);
+
+  let minimumDecimalPlaces = minimumDecimals;
+  let maximumDecimalPlaces = maximumDecimals;
+
+  const minBasedOnOrderOfMag = orderOfMagnitude > 2 ? 0 : 2;
+  if (orderOfMagnitude < 1) {
+    minimumDecimalPlaces = Math.max(minimumDecimals, significantDecimals);
+    maximumDecimalPlaces = Math.max(minBasedOnOrderOfMag, significantDecimals + 1);
+  } else {
+    minimumDecimalPlaces = minimumDecimals;
+    maximumDecimalPlaces = minBasedOnOrderOfMag;
+  }
+
+  return {
+    minimumDecimalPlaces,
+    maximumDecimalPlaces: Math.max(maximumDecimalPlaces + (precisionAdjustment ?? 0), minimumDecimalPlaces),
+  };
+}
+
+export function formatNative(value: string, defaultPriceValue: string, nativeSelected: SupportedCurrency[keyof SupportedCurrency]) {
+  'worklet';
+  if (!value) {
+    return defaultPriceValue || '';
+  }
+  if (value === 'undefined') {
+    return nativeSelected?.alignment === 'left' ? `${nativeSelected?.symbol}0.00` : `0.00 ${nativeSelected?.symbol}`;
+  }
```

Comment:
Not really an issue, but curious why we handle these two cases differently. Is it so that we don't ever flash `$0.00` on tokens with actual value?